boards/nordic/nrf52840dongle/nrf52840dongle_nrf52840.dts

```diff
@@ -48,14 +49,21 @@
 
 	pwmleds {
 		compatible = "pwm-leds";
-		red_pwm_led: pwm_led_0 {
-			pwms = <&pwm0 0 PWM_MSEC(20) PWM_POLARITY_INVERTED>;
+		ld1_green_pwm_led: ld1_green_led {
+			pwms = <&pwm0 3 PWM_MSEC(20) PWM_POLARITY_INVERTED>;
+			label = "LD1";
+			index = <0>;
+			color-mapping = <LED_COLOR_ID_GREEN>;
 		};
-		green_pwm_led: pwm_led_1 {
-			pwms = <&pwm0 1 PWM_MSEC(20) PWM_POLARITY_INVERTED>;
-		};
-		blue_pwm_led: pwm_led_2 {
-			pwms = <&pwm0 2 PWM_MSEC(20) PWM_POLARITY_INVERTED>;
+		ld2_rgb_pwm_led: ld2_rgb_led {
+			pwms = <&pwm0 0 PWM_MSEC(20) PWM_POLARITY_INVERTED>,
+			       <&pwm0 1 PWM_MSEC(20) PWM_POLARITY_INVERTED>,
+			       <&pwm0 2 PWM_MSEC(20) PWM_POLARITY_INVERTED>;
```

Comment:
Hi @nixward;

You should have come to Discord to discuss this before adding this feature...

See https://www.kernel.org/doc/Documentation/devicetree/bindings/leds/leds-pwm.yaml.

The maximum items of the `pwms` property is set to 1. So this PR breaks compatibility with the Linux `pwm-leds` DT binding. And a rule of Zephyr is to maintain DT compatibility with Linux...

See another discussion in https://github.com/zephyrproject-rtos/zephyr/pull/69227. My intention was to add the same functionality to the `led-pwm` driver right after, but I had to fall back...

So I think we will have to implement https://www.kernel.org/doc/Documentation/devicetree/bindings/leds/leds-pwm-multicolor.yaml.

Reply:
Hi @simonguinot,

Ok interesting, because the led_shell supported the properties provided by the led-controller binding it looked the right way to go and I unfortunately hadn't thought to check Linux bindings. If you haven't already started work on it I can have a look at creating a new driver to match the leds-pwm-multicolor binding. I'll drop in to Discord more often, thanks for the feedback.